lib/streamlit/elements/widgets/data_editor.py

```diff
@@ -203,6 +204,9 @@ def _parse_value(
         if column_data_kind == ColumnDataKind.BOOLEAN:
             return bool(value)
 
+        if column_data_kind == ColumnDataKind.DECIMAL:
+            return Decimal(str(value))
```

Comment:
I kind of assumed that it needs a string since I only have seen it being initialized with string. I also tried it with just passing in the function, and it seems to run into some weird issues.

Reply:
The `None` case is handled a bit above, so at that point it will never be `None`.